sr/comp/comp.py

```diff
@@ -36,6 +37,46 @@ def load_scorer(root: Path) -> ScorerType:
     return cast(ScorerType, score['Scorer'])
 
 
+def load_ranker(root: Path) -> RankerType:
+    """
+    Load the ranker module from Compstate repo.
+
+    :param Path root: The path to the compstate repo.
+    """
+
+    # Deep path hacks
+    score_directory = root / 'scoring'
+    ranker_source = score_directory / 'ranker.py'
+
+    if not ranker_source.exists():
+        # By default we support using the `league-ranker` package without
+        # modifications.
+        return ranker.LeagueRanker
+
+    saved_path = copy(sys.path)
+    sys.path.insert(0, str(score_directory))
+
+    try:
+        score = runpy.run_path(str(ranker_source))
+    finally:
+        sys.path = saved_path
```

Comment:
This does mean that delayed imports inside the ranker will break because the `sys.path` has been reverted by that point.

Reply:
True, this also impacts the scorer loading (and srcomp-scorer's use of converter modules); -> https://github.com/PeterJCLaw/srcomp/issues/42.